src/revwalk.rs

```diff
@@ -10,6 +10,19 @@ use crate::{raw, Error, Oid, Repository, Sort};
 pub struct Revwalk<'repo> {
     raw: *mut raw::git_revwalk,
     _marker: marker::PhantomData<&'repo Repository>,
+    hide_cb: Option<Box<dyn FnMut(Oid) -> bool>>,
+}
+
+extern "C" fn revwalk_hide_cb(commit_id: *const raw::git_oid, payload: *mut c_void) -> c_int {
+    unsafe {
```

Comment:
This'll want to get wrapped in `panic::call` to avoid aborting the process on panics

Reply:
Additionally `panic::check` will need to be called after C functions which may trigger the initial call of the callback

Reply:
Ok, I saw this in other places but did not really get it :)